rfcs/000-resource-notices.md

```diff
@@ -0,0 +1,88 @@
+- Feature Name: Resource Notices
+- Start Date: 2021-03-22
+- RFC PR: [amundsen-io/rfcs#0000](https://github.com/amundsen-io/rfcs/pull/0000) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Resource Notices
+
+## Summary
+
+Resource Notices allows Amundsen administrators to show informational, warning, and alert messages related to the different resources (tables, dashboards, people) we expose in Amundsen.
+
+## Motivation
+
+This feature will help administrators show messages related to deprecation, updates (or lack of), and informational messages related to specific resources. In Lyft's case, we use them to point final users to improved versions of current surfaced tables, so final users employ more trusted data.
+
+## Guide-level Explanation (aka Product Details)
+
+An Amundsen final user can see a notice in the table/dashboard/user detail page whenever their system admin adds one related to it.
+
+A notice is a small box with an icon and a message containing HTML markup (like links and bolded text).
+
+These will come in three flavors:
+
+- Informational: Marked with a blue "i" icon on the right side
+- Warning: Marked with an orange exclamation mark icon on the right side
+- Alert: Marked with a red exclamation mark icon on the right side
+
+For example, if company X wants to deprecate the use of one table, they can opt to add a new notice in their configuration file:
+
+```
+    ... Table Configuration
```

Comment:
does it only support config based notice instead of dynamic notice? E.g could the feature be used to show a table has data delay / quality issue which will require realtime metadata ingestion?

Reply:
this proposal is static only -- there is discussion / note that adding it to the graph (and thus realtime) is currently out-of-scope, but compatible with this change (there are use cases for each of these)

Reply:
Only config based for now.
That' a nice use case, we should consider it for the next phases of this feature!

Reply:
One more thought - what is the ballpark of notices we can add to config before performance starts to decay ? Would having 1k+ notices be reasonable ?

Reply:
Using the configuration it is a O(1) operation, so there is no issue in the number. It will look terrible though and it would break the balance on the design though.

Reply:
ok thanks, I meant 1 notice per table & add notices for 1k tables. In our case we have over 10k tables in Amundsen and a lot of them could use deprecation/pointing out to newer version.

Reply:
Oh, I see. In theory is fine, just a PITA to add! :)